Markdown tests fails after Marked update to 1.2.2

## 📝 Provide detailed reproduction steps (if any)

1.  Run `rm -rf node_modules yarn.lock && yarn` in the main repo
2.  Run `yarn test --files=markdown-gfm`

### ✔️ Expected result

Tests are passing

### ❌ Actual result

Tests do not pass.

```
FAILED TESTS:
  GFMDataProcessor
    todo lists
      ✖ should process todo lists
        Chrome 85.0.4183.121 (Linux x86_64)
      AssertionError: expected '<ul><li>[ ] Item 1</li><li>[x] Item 2</li></ul>' to equal '<ul><li><input disabled="" type="checkbox"></input>Item 1</li><li><input checked="" disabled="" type="checkbox"></input>Item 2</li></ul>'

      + expected - actual

      -<ul><li>[ ] Item 1</li><li>[x] Item 2</li></ul>
      +<ul><li><input disabled="" type="checkbox"></input>Item 1</li><li><input checked="" disabled="" type="checkbox"></input>Item 2</li></ul>
```

## 📃 Other details

*   Markedjs was updated yesterday: https://github.com/markedjs/marked/releases

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

--- a/packages/ckeditor5-markdown-gfm/package.json
+++ b/packages/ckeditor5-markdown-gfm/package.json
@@ -13,7 +13,7 @@
   "main": "src/index.js",
   "dependencies": {
     "ckeditor5": "^32.0.0",
-    "marked": "1.1.1",
+    "marked": "4.0.12",
     "turndown": "^6.0.0",
     "turndown-plugin-gfm": "^1.0.2"
   },

--- a/packages/ckeditor5-markdown-gfm/src/markdown2html/markdown2html.js
+++ b/packages/ckeditor5-markdown-gfm/src/markdown2html/markdown2html.js
@@ -7,7 +7,7 @@
  * @module markdown-gfm/markdown2html
  */
 
-import marked from 'marked';
+import { marked } from 'marked';
 
 // Overrides.
 marked.use( {
@@ -21,6 +21,14 @@ marked.use( {
 			// Remove bogus space after <input type="checkbox"> because it would be preserved
 			// by DomConverter as it's next to an inline object.
 			return Object.getPrototypeOf( this ).checkbox.call( this, ...args ).trimRight();
+		},
+
+		code( ...args ) {
+			// Since marked v1.2.8, every <code> gets a trailing "\n" whether it originally
+			// ended with one or not (see https://github.com/markedjs/marked/issues/1884 to learn why).
+			// This results in a redundant soft break in the model when loaded into the editor, which
+			// is best prevented at this stage. See https://github.com/ckeditor/ckeditor5/issues/11124.
+			return Object.getPrototypeOf( this ).code.call( this, ...args ).replace( '\n</code>', '</code>' );
 		}
 	}
 } );

--- a/packages/ckeditor5-markdown-gfm/tests/gfmdataprocessor/lists.js
+++ b/packages/ckeditor5-markdown-gfm/tests/gfmdataprocessor/lists.js
@@ -325,7 +325,7 @@ describe( 'GFMDataProcessor', () => {
 			);
 		} );
 
-		it( 'should create same bullet from different list indicators', () => {
+		it( 'should create different lists from different list indicators', () => {
 			testDataProcessor(
 				'* test\n' +
 				'+ test\n' +
@@ -333,13 +333,19 @@ describe( 'GFMDataProcessor', () => {
 
 				'<ul>' +
 					'<li>test</li>' +
+				'</ul>' +
+				'<ul>' +
 					'<li>test</li>' +
+				'</ul>' +
+				'<ul>' +
 					'<li>test</li>' +
 				'</ul>',
 
 				// After converting back list items will be unified.
 				'*   test\n' +
+				'\n' +
 				'*   test\n' +
+				'\n' +
 				'*   test'
 			);
 		} );

--- a/packages/ckeditor5-markdown-gfm/tests/gfmdataprocessor/strong-emphasis.js
+++ b/packages/ckeditor5-markdown-gfm/tests/gfmdataprocessor/strong-emphasis.js
@@ -33,10 +33,10 @@ describe( 'GFMDataProcessor', () => {
 			testDataProcessor(
 				'***This is strong and em.***',
 
-				'<p><strong><em>This is strong and em.</em></strong></p>',
+				'<p><em><strong>This is strong and em.</strong></em></p>',
 
 				// Normalized after converting back.
-				'**_This is strong and em._**'
+				'_**This is strong and em.**_'
 			);
 		} );
 
@@ -44,10 +44,10 @@ describe( 'GFMDataProcessor', () => {
 			testDataProcessor(
 				'Single ***word*** is strong and em.',
 
-				'<p>Single <strong><em>word</em></strong> is strong and em.</p>',
+				'<p>Single <em><strong>word</strong></em> is strong and em.</p>',
 
 				// Normalized after converting back.
-				'Single **_word_** is strong and em.'
+				'Single _**word**_ is strong and em.'
 			);
 		} );
 
@@ -55,10 +55,10 @@ describe( 'GFMDataProcessor', () => {
 			testDataProcessor(
 				'___This is strong and em.___',
 
-				'<p><strong><em>This is strong and em.</em></strong></p>',
+				'<p><em><strong>This is strong and em.</strong></em></p>',
 
 				// Normalized after converting back.
-				'**_This is strong and em._**'
+				'_**This is strong and em.**_'
 
 			);
 		} );
@@ -67,10 +67,10 @@ describe( 'GFMDataProcessor', () => {
 			testDataProcessor(
 				'Single ___word___ is strong and em.',
 
-				'<p>Single <strong><em>word</em></strong> is strong and em.</p>',
+				'<p>Single <em><strong>word</strong></em> is strong and em.</p>',
 
 				// Normalized after converting back.
-				'Single **_word_** is strong and em.'
+				'Single _**word**_ is strong and em.'
 			);
 		} );
 

